src/main/java/com/minecolonies/commands/RespawnCitizen.java

```diff
@@ -0,0 +1,121 @@
+    private static final String ENTITY_CITIZEN_NULL             = "Couldn't find entity of %d in %d";
+    private static final String COLONY_NULL                     = "Couldn't find colony %d";
+
+    public static final String DESC                             = "respawn";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public RespawnCitizen(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <CitizenId>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        final int colonyId = getIthArgument(args, 0, -1);
+        final int citizenId = getIthArgument(args, 1, -1);
+
+        //todo add this in a feature update when we added argument parsing and permission handling.
+        /*if(colonyId == -1)
+        {
+            colonyId = getColonyId(sender);
+        }*/
+
+        //No citizen or citizen defined.
+        if(colonyId == -1 || citizenId == -1)
+        {
+            sender.addChatMessage(new TextComponentString(String.format(NO_COLONY_CITIZEN_FOUND_MESSAGE, citizenId, colonyId)));
+            return;
+        }
+
+        //Wasn't able to get the citizen from the colony.
+        final Colony colony = ColonyManager.getColony(colonyId);
+        if(colony == null)
+        {
+            sender.addChatMessage(new TextComponentString(String.format(COLONY_NULL, colonyId)));
+            return;
+        }
+
+        final CitizenData citizenData = colony.getCitizen(citizenId);
+        if(citizenData == null)
+        {
+            sender.addChatMessage(new TextComponentString(String.format(CITIZEN_DATA_NULL, citizenId, colonyId)));
+            return;
+        }
+
+        //Wasn't able to get the entity from the citizenData.
+        final EntityCitizen entityCitizen = citizenData.getCitizenEntity();
+        if(entityCitizen == null)
+        {
+            sender.addChatMessage(new TextComponentString(String.format(ENTITY_CITIZEN_NULL, citizenId, colonyId)));
+            return;
+        }
+
+        sender.addChatMessage(new TextComponentString(String.format(CITIZEN_DESCRIPTION, entityCitizen.getEntityId(), entityCitizen.getName())));
+        final BlockPos position = entityCitizen.getPosition();
+        sender.addChatMessage(new TextComponentString(String.format(COORDINATES_XYZ, position.getX(), position.getY(), position.getZ())));
+
+        sender.addChatMessage(new TextComponentString(REMOVED_MESSAGE));
+
+        Log.getLogger().info("client? " + sender.getEntityWorld().isRemote);
+        //todo: make this threadsafe
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1135)